armada_jupyter/armada.py

```diff
@@ -39,13 +41,75 @@ def submit(submission: Submission):
             channel_credentials,
         )
 
-    client = ArmadaClient(channel)
+    client = parent_client(channel)
 
     queue = submission.queue
     job_set_id = submission.job_set_id
 
     # Create the PodSpec for the job
-    job_request_items = [create_armada_request(job, client) for job in submission.jobs]
-    client.submit_jobs(
+    job_request_items = [create_armada_request(job, client)]
+    resp = client.submit_jobs(
         queue=queue, job_set_id=job_set_id, job_request_items=job_request_items
     )
+
+    job_id = resp.job_response_items[0].job_id
+    return client, job_id
+
+
+def get_job_object(submission, job_id):
+    """
+    Returns the Job Objects with a matching job_id
+    """
+
+    for job in submission.jobs:
+        if job.job_id == job_id:
+            return job
+
+    return None
+
+
+def construct_url(job, job_id):
+    """
+    Constructs the URL for the Jupyter Notebook.
+
+    The layout of this URL is defined here:
+    https://github.com/G-Research/armada/blob/master/internal/
+    executor/util/kubernetes_object.go#L95
+    """
+
+    domain = os.environ.get("ARMADA_DOMAIN", "domain.com")
+    serviceport = job.podspec.containers[0].ports[0].containerPort
+    container_name = job.podspec.containers[0].name
```

Comment:
These fields aren't required to be set.  You should probably check for errors and maybe raise an exception.

Reply:
Do you think that should be done here or in another function?

Reply:
I'll leave that up to you. I try to follow SRP but a quick validation of one variable seems overkill.